x-pack/plugins/security_solution/public/resolver/view/submenu.tsx

```diff
@@ -50,10 +50,9 @@ export function compactNotationParts(num: number): [number, string, string] {
   if (!Number.isFinite(num)) {
     return [num, '', ''];
   }
-  let scale = 1;
-  while (scale < 1e12 && num / (scale * 1000) >= 1) {
-    scale *= 1000;
-  }
+
+  const scale = Math.pow(10, 3 * Math.min(Math.floor(Math.floor(Math.log10(num)) / 3), 4));
```

Comment:
:information_source: h/t @jonathan-buttner on the math

Reply:
![image](https://media.giphy.com/media/HBWbIuHvXI2Eo/giphy.gif)